src/operator/exhaustMap.ts

```diff
@@ -6,35 +6,33 @@ import {errorObject} from '../util/errorObject';
 import {OuterSubscriber} from '../OuterSubscriber';
 import {subscribeToResult} from '../util/subscribeToResult';
 
-export function exhaustMap<T, R, R2>(project: (value: T, index: number) => Observable<R>,
-                                     resultSelector?: (outerValue: T,
-                                                       innerValue: R,
-                                                       outerIndex: number,
-                                                       innerIndex: number) => R2): Observable<R> {
+export function exhaustMap<T, R, TResult>(project: (value: T, index: number) => Observable<R>,
```

Comment:
It's not clear looking at a variable declaration, for example, if `TResult` is a type parameter or a class/interface. Also, it's not consistent with the pattern we're using to name type parameters in this codebase as well as in RxJava.